cassiopeia/solver/dissimilarity_functions.py

```diff
@@ -52,22 +52,22 @@ def weighted_hamming_distance(
 
         if s1[i] != s2[i]:
             if s1[i] == 0 or s2[i] == 0:
-                if w:
+                if weights:
                     if s1[i] != 0:
-                        d += w[i][s1[i]]
+                        d += weights[i][s1[i]]
                     else:
-                        d += w[i][s2[i]]
+                        d += weights[i][s2[i]]
                 else:
                     d += 1
             else:
-                if w:
-                    d += w[i][s1[i]] + w[i][s2[i]]
+                if weights:
+                    d += weights[i][s1[i]] + weights[i][s2[i]]
                 else:
                     d += 2
         else:
             if s1[i] != 0:
-                if w:
-                    d -= 2 * w[i][s1[i]]
+                if weights:
+                    d -= 2 * weights[i][s1[i]]
 
```

Comment:
I actually think we should remove this functionality. IMO we should only be penalizing differences here (i.e., tallying dissimilarities). 

In the similarity functions this will obviously be present.